### DISCUSSION TITLE
AnalyticalTable: Selecting checkbox based on value from model

### DISCUSSION THREAD

**Discussion by shivamdz** [UI5/webcomponents-react#3049] [Q&A] [answered]
Hi Everyone,

We are using Analytical Table to display tree data and we want to select row based on field in data.
So, i have a data model which looks like below format (it is nested data) :
[{
name: "john",
age: 33
isSelected : true
}, {
name: "doe",
age: 44
isSelected : false
} ]

On Average it would have around 5-6k records.

Now based on 'isSelected' field i want to select the rows .One option is to use "selectedRowIds" property but this need objects in {'1': true, '1.1.2' : true} format and i need to parse above data model to get the required format (One additional loop)

I am looking for a simpler way ( like "useRowDisableSelection" hook which is provided for disabling the row ) to select and un select based on condition.

Can anyone recommend the best way to achieving this ?

Thanks,
Shivam

**Comment by Lukas742** [comment, marked as answer]
Hi @shivamdz 

you can create your own plugin hook to enable this behavior. [codeSandbox](https://codesandbox.io/s/treetable-forked-47ovee?file=/src/App.js)

**Reply by Lukas742** [reply]
With[ v0.25.2](https://github.com/SAP/ui5-webcomponents-react/releases/tag/v0.25.2) you can now import the `useManualRowSelect` plugin hook to get this behavior out of the box.
